### PR TITLE
Test moving aliases on room upgrades

### PR DESCRIPTION
**Pull request by richvdh**
Builds on #513. Tests for https://github.com/matrix-org/synapse/pull/4101.

### CODE REVIEW

File: tests/30rooms/60version_upgrade.pl [hunk 1]
```diff
@@ -51,6 +52,7 @@ sub upgrade_room {
          new_version => $new_version,
       },
    )->then( sub {
+
```
Comment (erikjohnston) [inline tests/30rooms/60version_upgrade.pl]: (Did you mean to add these new lines?)

File: tests/30rooms/60version_upgrade.pl [hunk 2]
```diff
@@ -285,6 +287,124 @@ sub upgrade_room_synced {
...
+               foreach my $room_id ( $room_id, $new_room_id ) {
+                  my $rs = $received_events{$room_id} // {};
+                  foreach ( qw( m.room.aliases m.room.canonical_alias ) ) {
+                     log_if_fail "$_ in $room_id", $rs->{$_};
+                     return 0 unless $rs->{$_};
+                  }
+               }
+               return 1;
+            },
+         );
```
Comment (erikjohnston) [inline tests/30rooms/60version_upgrade.pl]: (This feels sad the amount of boiler plate code here)

File: tests/30rooms/60version_upgrade.pl [hunk 3]
```diff
@@ -384,55 +381,50 @@ sub upgrade_room_synced {
...
       })->then( sub {
-         my ( $new_room_id, $sync_body ) = @_;
+         ( $new_room_id ) = @_;
 
-         my $old_aliases = $received_events{$room_id}->{'m.room.aliases'}->{$server_name};
-         assert_deeply_eq( $old_aliases->{content}, {aliases => []}, "aliases on old room" );
+         matrix_get_room_state(
+            $creator, $room_id,
+            type=>'m.room.aliases', state_key=>$server_name,
+         );
```
Comment (erikjohnston) [inline tests/30rooms/60version_upgrade.pl]: Any reason for using `matrix_get_room_state` rather than just syncing?

Comment (richvdh) [reply tests/30rooms/60version_upgrade.pl]: I decided it was going to be a pain to find the right events in the sync. Could change it though.

Comment (erikjohnston) [reply tests/30rooms/60version_upgrade.pl]: Nah, its fine

**Comment by richvdh** [comment]
sigh. retest this please.

**Pull request merged**